src/mount/conf.cc

```diff
@@ -53,8 +54,11 @@ extern "C" void mount_ceph_get_config_info(const char *config_file,
   for (const auto& mon : monc.monmap.addr_mons) {
     auto& eaddr = mon.first;
 
-    // For now, kernel client only accepts legacy addrs
-    if (!eaddr.is_legacy())
+    /*
+     * Skip v2 addrs if we're running in legacy ms_mode. Skip
+     * v1 addrs for any other mode.
+     */
+    if (v2_addrs == eaddr.is_legacy())
```

Comment:
This is a bit cryptic IMO and also potentially includes `NONE` and `ANY` addresses.  I'd go with something more precise, e.g. `if ((!v2_addrs && !eaddr.is_legacy()) || (v2_addrs && !eaddr.is_msgr2()))` or two separate `if`s.

Reply:
Thanks, that makes sense. I just pushed an update to do that.